### PR TITLE
Update gql types for storefront

### PR DESCRIPTION
**Pull request by piotrgrundas**
Needed for [Enable incremental server-side regeneration for store pages](https://github.com/mirumee/saleor-storefront/pull/988)

### CODE REVIEW

File: src/api/products/ProductList.ts [hunk 1]
```diff
@@ -14,14 +19,38 @@ export class ProductList extends BaseList<
...
+  getTotalCount: GetTotalCount<ProductListQuery> = result =>
+    result.data.products?.totalCount!;
+
   mapQueryData: MapQueryData<
     ProductListQuery,
     ProductList_products_edges_node
   > = data => data.products?.edges.map(({ node }) => node);
 
-  query = (variables: ProductListVariables) =>
+  query = async (variables: ProductListVariables) =>
```
Comment (jwm0) [inline src/api/products/ProductList.ts:30]: looks like some leftover along with the comment below

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/api/products/ProductList.ts (hunk 1 above)
Comment (piotrgrundas) [reply src/api/products/ProductList.ts:30]: I was sure I removed this... 🙄  My bad!

File: src/api/products/ProductList.ts [hunk 2]
```diff
@@ -1,4 +1,9 @@
-import BaseList, { GetPageInfo, MapQueryData } from "../../helpers/BaseList";
+// import { ApolloQueryResult } from "apollo-client";
```
Comment (mmarkusik) [inline src/api/products/ProductList.ts:1]: Needs removing?

**Review by mmarkusik — APPROVED** [review]

**Review by dominik-zeglen — APPROVED** [review]
Lgtm besides code cleanup

**Review by jwm0 — APPROVED** [review]

**Pull request merged**